protocols/kad/src/behaviour.rs

```diff
@@ -127,8 +127,9 @@ impl Default for KademliaConfig {
 impl KademliaConfig {
     /// Sets a custom protocol name.
     ///
-    /// Kademlia nodes only communicate with other nodes using the same protocol name. Using a
-    /// custom name therefore allows to segregate the DHT from others, if that is desired.
+    /// Kademlia nodes only communicate with other nodes using the same protocol
+    /// name. Using a custom name therefore allows to segregate the DHT from
+    /// others, if that is desired.
```

Comment:
Out of scope of this PR, but it just crossed my mind that this isn't really very effective as a means for segregation, since this protocol name is negotiated on substreams, i.e. after the peer may have already been added to the routing table, if I'm not mistaken.

Reply:
Yes, exactly. The protocol name is only negotiated on substreams after the connection is already established.

Reply:
Tracked in https://github.com/libp2p/rust-libp2p/issues/1611.